src/DGtal/geometry/curves/estimation/LambdaMST2D.ih

```diff
@@ -0,0 +1,117 @@
+
+template < typename TSpace, typename TSegmentation, typename Functor >
+inline
+void
+DGtal::LambdaMST2DEstimator< TSpace, TSegmentation, Functor >::init ( const ConstIterator & itb, const ConstIterator & ite )
+{
+  myBegin = itb;
+  myEnd = ite;
+}
+
+template < typename TSpace, typename TSegmentation, typename Functor >
+inline
+void
+DGtal::LambdaMST2DEstimator< TSpace, TSegmentation, Functor >::attach ( const TSegmentation & segmentComputer )
+{
+  dssSegments = &segmentComputer;
+}
+
+template < typename TSpace, typename TSegmentation, typename Functor >
+inline
+bool
+DGtal::LambdaMST2DEstimator< TSpace, TSegmentation, Functor >::isValid () const
+{
+  return ( dssSegments != 0 );
+}
+
+template < typename TSpace, typename TSegmentation, typename Functor >
+inline
+typename TSpace::RealVector
+DGtal::LambdaMST2DEstimator< TSpace, TSegmentation, Functor >::eval ( const typename TSpace::Point & point )
+{
+  assert ( dssSegments != 0 );
+  typename TSegmentation::SegmentComputerIterator DSS = dssSegments->begin();
+  typename TSegmentation::SegmentComputerIterator lastDSS = dssSegments->end();
+  Value tangent, partial;
+  for ( ; DSS != lastDSS; ++DSS )
+  {
+    if ( DSS->isInDSS ( point ) )
+    {
+      unsigned int pos = std::distance ( DSS.begin(), std::find ( DSS.begin(), DSS.end(), point ) );
+      unsigned int dssLen = std::distance ( DSS.begin(), DSS.end() );
+      SegmentComputer comp ( *DSS );
+      partial = myFunctor ( comp, pos, dssLen );
+      tangent += partial;
+    }
+  }
+  if ( tangent.second != 0. )
+    return tangent.first / tangent.second;
+  else
+    return tangent.first;
+}
+
+template < typename TSpace, typename TSegmentation, typename Functor >
+template < typename Containter >
+inline
+void
+DGtal::LambdaMST2DEstimator< TSpace, TSegmentation, Functor >::eval ( std::back_insert_iterator < Containter > result )
+{
+  assert ( myBegin != myEnd && isValid() );
+  std::vector < Value > outValues ( std::distance ( myBegin, myEnd ) );
+  typename TSegmentation::SegmentComputerIterator DSS = dssSegments->begin();
+  typename TSegmentation::SegmentComputerIterator lastDSS = dssSegments->end();
+  for ( ; DSS != lastDSS; ++DSS )
+  {
+    unsigned int dssLen = std::distance ( DSS.begin(), DSS.end() );
+    SegmentComputer comp ( *DSS );
+    for ( unsigned int i = 0; i < dssLen; i++ )
+      outValues[ std::distance ( myBegin, DSS.begin() ) + i ] += myFunctor ( comp, i, dssLen );
+  }
+  accumulate ( outValues, result );
+}
+
+template < typename TSpace, typename TSegmentation, typename Functor >
+template < typename Containter >
+inline
+void
+DGtal::LambdaMST2DEstimator< TSpace, TSegmentation, Functor >::accumulate ( std::vector < Value > & outValues,
+									    std::back_insert_iterator < Containter > result )
+{
+  for ( unsigned int i = 0; i < outValues.size(); i++ )
+  {
+    Value & tangent = outValues[i];
+    if ( tangent.second != 0. )
+      result = ( tangent.first / tangent.second );
```

Comment:
I remember learning to write: *result++ = ...
but you are right, this is correct.